<!--
In laptop
Launch this file
rosrun line_detection line_detection node
rostopic pub /mission/action_state msgs/IntSmped to 1 
rostopic pu/reg_ratio msgs/FloatArrayolStamped to [0-1]
rostopic pu-r 100 /fmSafe/actuation_enable msgs/BoololStamped to True

In the frobit




 -->


<launch> 
 <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video1" />   
    <param name="image_width" value="1280" />
    <param name="image_height" value="720" />
    <param name="pixel_format" value="mjpeg" />
    <param name="camera_frame_id" value="/camera_link" />
    <param name="framerate" value="30" />
    <param name="contrast" value="32" />
    <param name="brightness" value="32" />
    <param name="saturation" value="32" />
    <param name="autofocus" value="false" />
    <param name="focus" value="51" />
    <!-- <param name="do_rectify" value="true" /> not working --> 
    <param name="camera_info_url" value="package://line_detection/config/head_camera.yaml" /> 
    <param name="camera_name" value="head_camera" />
    <param name="io_method" value="mmap"/>
  </node>

 <!--<node name="line_detector" pkg="line_detection" type="line_detection_node" output="screen" > 
 </node>-->

 <node name="line_pid_control" pkg="line_pid" type="line_pid_node" output="screen" > 
    <param name="update_rate" value="20"/>
    <param name="drive_kp" value="5.0"/>
    <param name="drive_ki" value="0.01"/>
    <param name="drive_max_i" value="3" />
    <param name="drive_kd" value="2.0"/>
    <param name="drive_feed_forward" value="0.00"/>
    <param name="drive_max_output" value="0.4"/>
    <param name="forward_speed" value="0.5" />
    <param name="line_sub" value="/line_detector/perception/line" />
    <param name="action_topic" value="/mission/action_state" />          
    <param name="pid_debug_pub" value="/fmInformation/pid" />
    <param name="command_pub" value="/fmCommand/cmd_vel" />
 </node>

 <node pkg="safety_speed_node" name="safety_speed" type="safety_speed_node.py" output="screen">
    <param name="vel_topic" value="/fmCommand/cmd_vel"/>
    <param name="safe_vel_topic" value="/safe_vel"/>
    <param name="reg_ratio_topic" value="/reg_ratio"/>
 </node>

</launch>